### PR TITLE
Diagonal walls and secret doors

### PR DESCRIPTION
**Pull request by ErhardSteinhauer**
## About the PR
New entities:
- Diagonal walls: wood, reinforced and uranium.
- Secret doors (can be constructed): wood, reinforced, shuttle, uranium.

## Why / Balance
More structures for mappers and smugglers.

## Technical details
new .yml and .rsi

## Media
image
Fig. 1 - Structures (secret doors aren't present since they look like walls anyways, notice boards were moved to separate PR).
- [X] I have added screenshots/videos to this PR showcasing its changes ingame.

**Changelog**
:cl: erhardsteinhauer
- add: Added buildable entities: secret doors (wood, reinforced, shuttle, uranium). And some diagonal walls for mappers.

### CODE REVIEW

File: Resources/Prototypes/_NF/Entities/Structures/Walls/diagonal_walls.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/_NF/Entities/Structures/Walls/diagonal_walls.yml:1]: WallShuttleDiagonal should not be a parent of Diagonals walls, this parent got reflective

**Review by dvir001 — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/_NF/Entities/Structures/Walls/grille_diagonal.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/_NF/Entities/Structures/Walls/grille_diagonal.yml:1]: Same issue:

WallShuttleDiagonal should not be a parent of Diagonals walls, this parent got reflective

**Review by FoxxoTrystan — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/_NF/Entities/Structures/Walls/diagonal_walls.yml
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Structures/Walls/diagonal_walls.yml:1]: > WallShuttleDiagonal should not be a parent of Diagonals walls, this parent got reflective

It is parent for solid diagonal wall in the base game. I can explicitly specify the reflective component as 0.

File: Resources/Prototypes/_NF/Entities/Structures/Doors/secret_door.yml
[the remark this answers is not in the document]
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Structures/Doors/secret_door.yml:1]: Sure

File: Resources/Prototypes/_NF/Entities/Structures/Windows/window.yml
[the remark this answers is not in the document]
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Structures/Windows/window.yml:1]: My bad thought that it from my previous iteration of this branch

**Comment by ErhardSteinhauer** [comment]
Changed a few things around:
- Kegs moved to separate PR.
- Notice board moved to separate PR.
- Changed parent on diagonal windows from `BaseItem` to `Window`.
- Changed parent on diagonal walls from `WallShuttleDiagonal` to `WallSolidDiagonal`.
- Removed resistances from secret doors.
- Restored `windows.yml`.

**Comment by ErhardSteinhauer** [comment]
Made so diagonal windows inherit from their respective base variations: eg., diagonal plasma window inherits from plasma window etc.

**Comment by dvir001** [comment]
I made multi parents changed to walls, all seem fine to me after testing, please confirm on your side its all good.

**Review by dvir001 — APPROVED** [review]

**Comment by ErhardSteinhauer** [comment]
> I made multi parents changed to walls, all seem fine to me after testing, please confirm on your side its all good.

Your version works but I made some changes too and those appear to work fine on my end:
- Added new entity to use as a parent instead of WallSolidDiagonal (which inherits from WallShuttleDiagonal and has reflect component) - BaseWallDiagonal.
- Replaced WallSolidDiagonal with BaseWallDiagonal everywhere.

**Comment by ErhardSteinhauer** [comment]
Also what's up with Map Prototype Checker throwing errors?

**Comment by ErhardSteinhauer** [comment]
Moving diagonal windows and grille to a separate PR, leaving here here diagonal walls and secret doors.

**Review by FoxxoTrystan — APPROVED** [review]

**Pull request merged**